pkg/controller/importconfig/cluster_info.go

```diff
@@ -74,6 +75,16 @@ func getBootstrapKubeConfigDataFromImportSecret(ctx context.Context, clientHolde
 		return nil, nil, nil
 	}
 
+	// check if the current context cluster name of the bootstrap kubeconfig is changed
+	validCtxClusterName, err := validateContextClusterName(ctx, clientHolder.RuntimeClient, ctxClusterName)
```

Comment:
note: why we validate here? all generation and validation should be in bootstrap package.

Reply:
this is to fix YangLe's comment https://github.com/stolostron/managedcluster-import-controller/pull/346#issuecomment-2126303203

Reply:
yeah I see. I mean the whole validation func should be put in the bootstrap package.

Reply:
I will submit a follow-up PR to refactor this, and keep the current PR changes as small as possible to facilitate cherrypick to the old releases.